applications/visualizer/frontend/src/components/viewers/EM/EMStackTilesViewer.tsx

```diff
@@ -213,77 +263,44 @@ const EMStackViewer = () => {
       interactions: interactions,
     });
 
-    ringEM.current = new SlidingRing({
+    ringEM.current = new SlidingLayer({
+      map: map,
       cacheSize: ringSize,
       startAt: startSlice,
       extent: [minSlice, maxSlice],
-      onPush: (slice) => {
-        const layer = newEMLayer(firstActiveDataset, slice, tilegrid, projection);
-        layer.setOpacity(0);
-        map.addLayer(layer);
-        return layer;
-      },
-      onSelected: (_, layer) => {
-        layer.setOpacity(1);
-      },
-      onUnselected: (_, layer) => {
-        layer.setOpacity(0);
-      },
-      onEvict: (_, layer) => {
-        map.removeLayer(layer);
-      },
+      newLayer: (slice) => newEMLayer(firstActiveDataset, slice, tilegrid, projection),
     });
 
-    ringSeg.current = new SlidingRing({
+    ringSeg.current = new SlidingLayer({
+      map: map,
       cacheSize: ringSize,
       startAt: startSlice,
       extent: [minSlice, maxSlice],
-      onPush: (slice) => {
-        const layer = newSegLayer(firstActiveDataset, slice);
-        layer.setOpacity(0);
+      newLayer: (slice) => newSegLayer(firstActiveDataset, slice),
+      onSlide: (slice, layer) => {
         layer.setStyle((feature) => neuronsStyleRef.current(feature));
-        map.addLayer(layer);
-        return layer;
-      },
-      onSelected: (slice, layer) => {
-        layer.setOpacity(1);
         currSegLayer.current = layer;
         segSetSlice(slice);
       },
-      onUnselected: (_, layer) => {
-        layer.setOpacity(0);
-      },
-      onEvict: (_, layer) => {
-        map.removeLayer(layer);
-      },
     });
 
-    map.on("click", (e) => onNeuronSelectRef.current(e.coordinate));
+    map.on("click", (e) => onFeatureClickRef.current(e.coordinate));
 
-    ringSynSeg.current = new SlidingRing({
+    ringSynSeg.current = new SlidingLayer({
+      map: map,
       cacheSize: ringSize,
       startAt: startSlice,
       extent: [minSlice, maxSlice],
-      onPush: (slice) => {
-        const layer = newSynapsesSegLayer(firstActiveDataset, slice);
-        layer.setOpacity(0);
-        layer.setStyle({
-          "fill-color": "blue",
-          "stroke-color": "blue",
+      newLayer: (slice) => newSynapsesSegLayer(firstActiveDataset, slice),
+      onSlide: (_, layer) => {
+        layer.setStyle((feature) => {
+          const isSelected = feature.get("selected");
```

Comment:
Same here, we shouldn't add special attributes, but use the synchronizers, and if we need to differenciate between neurons and synapses, it would be better to simply check if the "name" of the feature is part of the list of neurons that are loaded (or from the whole list of neurons that is loaded by workspace)